### PR TITLE
Feature 1 user login

### LINKED ISSUE #66: [Fix Feature 1] - refactoring and debugging
- Fix bug where the user could not be read from the database
- Refactor so that clean architecture is implemented

### PR DESCRIPTION
**Pull request by swaner2**
# Description
Fixes #66 

Refactored log in use case so that the presenter is not directly dependent on the view and added comments, javadocs throughout for clarity. Also fixed bug where a user could not be retrieved from the database after a user signs in. 

Next step is to write unit tests. 

# Checklist
- [x] My PR does not contain unrelated files that may cause conflicts
- [x] I have performed a self-review of my own code
- [x] I have commented my code with javadoc and perhaps in-line comments (particularly in
  hard-to-understand areas)
- [x] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have not violated clean architecture principles
- [x] I have opted to use the most general form of datastructures (example: use `List` instead of
  `ArrayList` wherever possible)

### CODE REVIEW

File: app/src/main/java/com/group80/uoftinder/login_use_case/LoginPresenterFormatter.java [hunk 1]
```diff
@@ -1,55 +1,49 @@
...
     }
 
     /**
      * If email and password combination do not match, pop-ups a message indicating login failure
-     * @param error error message
      */
     @Override
-    public void prepareFailureViewLogin(String error) {
-        Toast.makeText((Context) loginViewInterface, error, Toast.LENGTH_SHORT).show();
+    public void prepareLoginFailureView(String unsuccessful) {
```
Comment (helen-li) [inline app/src/main/java/com/group80/uoftinder/login_use_case/LoginPresenterFormatter.java:32]: The javadoc param tag for `prepareLoginFailureView` and for `prepareSuccessView` don't match here - should be two quick fixes

**Review by helen-li — COMMENTED** [review]
Looks super good and clean! Just dropped one small comment regarding javadoc

**Review by helen-li — APPROVED** [review]
Especially love the class javadocs that clarify the clean architecture layers! Super good work Jess

**Review by LanceLi1416 — APPROVED** [review]
Looks good!

**Review by rsrinivasan1 — APPROVED** [review]
Looks good! Glad that you made the login use case clean.

**Pull request merged**